pex/interpreter.py

```diff
@@ -95,7 +95,7 @@ def from_id_string(cls, id_string):
   @classmethod
   def from_path(cls, dirname):
     interp, version = dirname.split('-')
-    major, minor, patch = version.split('.')
+    major, minor, patch = version.split('.')[:3]
```

Comment:
I think the larger question here is why the interpreter discovery mechanism is finding what it deems valid interpreters that look like 'CPython-2.7.3.tmp.ba3a2c39daab46ef978ba07fe335079d' or where they're coming from to begin with. 

any clues from your analysis? here's what these should normally look like in cache:

```
$ find ~/.pex/interpreters
/Users/kw/.pex/interpreters
/Users/kw/.pex/interpreters/CPython-2.7.10
/Users/kw/.pex/interpreters/CPython-2.7.10/wheel
/Users/kw/.pex/interpreters/CPython-2.7.10/wheel-0.24.0-py2.7.egg
/Users/kw/.pex/interpreters/CPython-2.7.6
/Users/kw/.pex/interpreters/CPython-2.7.6/wheel
/Users/kw/.pex/interpreters/CPython-2.7.6/wheel-0.24.0-py2.7.egg
```

but in your interpreter cache, you'd probably find the `*.tmp.*` variety which is likely where this code is choking.

related:
- https://github.com/pantsbuild/pex/issues/140
- https://github.com/pantsbuild/pants/issues/2038